base/libgit2/callbacks.jl

```diff
@@ -110,13 +128,16 @@ function authenticate_ssh(creds::SSHCredentials, libgit2credptr::Ptr{Ptr{Void}},
             passdef = creds.pass # check if credentials were already used
             if passphrase_required && (isempty(passdef) || isusedcreds)
                 if Sys.iswindows()
-                    passdef = Base.winprompt(
+                    response = Base.winprompt(
                         "Your SSH Key requires a password, please enter it now:",
                         "Passphrase required", privatekey; prompt_username = false)
-                    isnull(passdef) && return Cint(Error.EAUTH)
-                    passdef = Base.get(passdef)[2]
+                    isnull(response) && return user_abort()
+                    passdef = unsafe_get(response)[2]
                 else
-                    passdef = prompt("Passphrase for $privatekey", password=true)
+                    response = prompt("Passphrase for $privatekey", password=true)
+                    isnull(response) && return user_abort()
+                    passdef = unsafe_get(response)
+                    isempty(passdef) && return user_abort()  # Ambiguous if EOF or newline
```

Comment:
I did some digging into this.

For SSH having a empty passphrase is considered as not having any passphrase at all. Since we can tell the difference between SSH keys that do and do not require passphrases (encrypted private keys have second line as "Proc-Type: 4,ENCRYPTED") and only prompt when a private key is encrypted it should be safe to abort if a empty passphrase is entered at the prompt.

For HTTP I ended up doing some experimentation with [httpbin](https://github.com/kennethreitz/httpbin) and using empty passwords for both HTTP Basic and Digest authentication. Both seem to consider an empty password invalid and will re-prompt for credentials.